File: src/core/Stack.ts
/**
 * A generic stack that supports any type of class, T.
 */
export class Stack<T> {
    private items: T[] = []

    push(item: T) {
        this.items.push(item)
    }

    pop(): T | undefined {
        return this.items.pop()
    }

    peek(): T | undefined {
        return this.items[this.items.length - 1]
    }

    isEmpty(): boolean {
        return this.items.length == 0
    }
}

File: src/core/Stack.py
class Stack():
    def __init__(self) -> None:
        self.items = []

    def push(self, item):
        self.items.append(item)

    def pop(self):
        return self.items.pop()

    def peek(self):
        return self.items[-1]

    def clear(self):
        while not self.isEmpty():
            self.pop()

    def isEmpty(self) -> bool:
        return len(self.items) == 0